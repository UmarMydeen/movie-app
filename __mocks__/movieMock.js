const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster',
    imdbRating: '9.2',
    Poster: 'mockUrl'
}

export default movieMock;