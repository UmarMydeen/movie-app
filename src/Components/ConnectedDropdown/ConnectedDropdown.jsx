import React from 'react';
import PropTypes from 'prop-types';

const ConnectedDropdown = ({
    label,
    options,
    onSelection
}) => {
    return (
        <div class="input-group input-group-sm mb-3">
            <div class="input-group-prepend">
                <span class="input-group-text" id="inputGroup-sizing-sm">
                    {label}
                </span>
            </div>
            <select
                class="form-select"
                aria-label="Default select example"
                onChange={onSelection}
            >
                <option selected value="">--Select--</option>
                {
                    options.map(({ value, text }) => (
                        <option value={value}>{text}</option>
                    ))
                }
            </select>
        </div>
    );
}

ConnectedDropdown.propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    onSelection: PropTypes.func.isRequired,
};

export default ConnectedDropdown;