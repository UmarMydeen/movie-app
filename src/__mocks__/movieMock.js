const movieMock = {
    imdbID: 'mockimdbID1',
    Title: 'mockTitle',
    Poster: 'mockPoster'
}

export default movieMock;