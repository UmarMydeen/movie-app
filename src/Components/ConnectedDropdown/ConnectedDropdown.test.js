import { fireEvent, getByTestId, render, screen } from '@testing-library/react';
import ConnectedDropdown from './ConnectedDropdown';
import movieMock from '../../../__mocks__/movieMock';

let mockProps;

beforeEach(() => {
    mockProps = {
        label: 'mockLabel',
        options: [{
            text: 'mockText',
            value: 'mockValue'
        }],
        onSelection: jest.fn()
    }
})

test('should render component correctly', () => {
    const component = render(<ConnectedDropdown {...mockProps} />);
    expect(component).toMatchSnapshot();
});

test('should trigger onSelection method selection', () => {
    const component = render(<ConnectedDropdown {...mockProps} />);
    fireEvent.change(component.getByTestId('connectedDropdown'));
    expect(mockProps.onSelection).toHaveBeenCalled();
});