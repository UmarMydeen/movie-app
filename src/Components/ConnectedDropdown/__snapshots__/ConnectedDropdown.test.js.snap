// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input-group input-group-sm mb-3"
      >
        <div
          class="input-group-prepend"
        >
          <span
            class="input-group-text"
            id="inputGroup-sizing-sm"
          >
            mockLabel
          </span>
        </div>
        <select
          aria-label="Default select example"
          class="form-select"
          data-testid="connectedDropdown"
        >
          <option
            value=""
          >
            --Select--
          </option>
          <option
            value="mockValue"
          >
            mockText
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input-group input-group-sm mb-3"
    >
      <div
        class="input-group-prepend"
      >
        <span
          class="input-group-text"
          id="inputGroup-sizing-sm"
        >
          mockLabel
        </span>
      </div>
      <select
        aria-label="Default select example"
        class="form-select"
        data-testid="connectedDropdown"
      >
        <option
          value=""
        >
          --Select--
        </option>
        <option
          value="mockValue"
        >
          mockText
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
